### PR TITLE
Add AWS read timeout to prevent random SNS delays from causing problems

### PR DESCRIPTION
**Pull request by maroux**

### CODE REVIEW

File: aws_client.go [hunk 1]
```diff
@@ -179,7 +180,9 @@ func (a *awsClient) PublishSNS(ctx context.Context, settings *Settings, messageT
 			TopicArn:          &topic,
 			Message:           &payload,
 			MessageAttributes: attributes,
-		})
+		},
+		request.WithResponseReadTimeout(settings.AWSReadTimeoutS),
```
Comment (mngo87) [inline aws_client.go]: this has to be fed into the sns call

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ before_script:
 - go vet $EXCLUDE_VENDOR
 - misspell -error -locale US *.go
 - megacheck $EXCLUDE_VENDOR
-- for file in $EXCLUDE_VENDOR; do diff -u <(echo -n) <(gofmt -d -s $file); done
+- for file in $EXCLUDE_VENDOR; do diff -u <(echo -n) <(gofmt -d -s $file) || travis_terminate 1; done
```
[the remark this answers is not in the document]
Comment (maroux) [reply .travis.yml]: It doesn't fail without it since `set -e` isn't set by travis by default.

File: aws_client.go (hunk 1 above)
Comment (maroux) [reply aws_client.go]: It is right? Not sure what you mean..

Comment (mngo87) [reply aws_client.go]: my mistake, i just read that line incorrectly

**Pull request merged**